actions/src/main/kotlin/com/alfresco/content/actions/ProcessDefinitionsViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.alfresco.content.actions
+
+import android.content.Context
+import com.airbnb.mvrx.Fail
+import com.airbnb.mvrx.MavericksViewModel
+import com.airbnb.mvrx.MavericksViewModelFactory
+import com.airbnb.mvrx.Success
+import com.airbnb.mvrx.ViewModelContext
+import com.alfresco.content.data.Entry
+import com.alfresco.content.data.TaskRepository
+import com.alfresco.content.data.payloads.LinkContentPayload
+import com.alfresco.coroutines.asFlow
+import kotlinx.coroutines.launch
+
+internal class ProcessDefinitionsViewModel(
+    state: ProcessDefinitionsState,
+    val context: Context
+) : MavericksViewModel<ProcessDefinitionsState>(state) {
+
+    init {
+        buildModel()
+    }
+
+    private fun buildModel() = withState { state ->
+        viewModelScope.launch {
+            linkContentToProcess(state.entry).execute {
+                when (it) {
+                    is Success -> {
+                        copy(linkContent = it)
+                    }
+                    is Fail -> {
+                        copy(linkContent = it)
+                    }
+                    else -> {
+                        copy(linkContent = it)
```

Comment:
Can we use it without any condition as its executing the same code in all the three conditions?

Reply:
there would be success and failure cases. On the basis of that condition, the next functionality will be executed. So made 3 cases and return the same request only.